aldryn_reversion/tests/test_admin_views.py

```diff
@@ -0,0 +1,189 @@
+        """
+        Mock a post request to admin view. Returns result ofadmin recover_view
+        processing (response) for given object (object_or_version).
+        If object is a Version - uses version.object_version.object.
+        If version_id was not provided - latest available version.id is used.
+        """
+        if version_id is None:
+            version_id = str(self.get_version(obj_or_version).id)
+        url = self.get_admin_url_for_obj(self.get_object(obj_or_version))
+        request = self.get_su_request(language, url)
+        request.method = 'POST'
+        admin_instance = self.get_admin_instance_for_object(obj_or_version)
+        return admin_instance.recover_view(request, version_id)
+
+    def check_info(self, response, info_string, versions=None):
+        """
+        Check that response contains info_string message,
+        if versions are passed builds version info strings and checks that
+        they were rendered.
+        """
+        self.assertContains(response, info_string)
+        if versions:
+            search_for = [
+                PLACEHOLDER_STR.format(self.build_string_args(version))
+                for version in versions]
+            for info in search_for:
+                self.assertContains(response, info)
+
+    def test_recover_view_access(self):
+        # test accessible for admin registered objects, should be greated than 4
+        # characters, since '/en/' is 4, and actually not raises an exception
+        self.assertGreater(
+            len(self.get_admin_url_for_obj(self.simple_registered)),
+            4)
+        # test not accessible for not registered objects
+        with self.assertRaises(NoReverseMatch):
+            self.get_admin_url_for_obj(self.simple_no_admin)
+
+    def test_recover_view_works_with_simple_objects(self):
+        simple_version = get_latest_version_for_object(self.simple_registered)
+        self.simple_registered.delete()
+        self.assertEqual(SimpleRegistered.objects.count(), 0)
+
+        response = self.get_recover_view_response(simple_version)
+        for info in ALL_INFO_MESSAGES:
+            self.assertNotContains(response, info)
+
+        # test recover button is accessible for simple object
+        self.assertContains(response, RECOVER_BUTTON)
+        # test recover is actually recovers object
+        response = self.post_recover_view_response(simple_version)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(SimpleRegistered.objects.count(), 1)
+
+    def test_recover_warnings_on_conflicts(self):
+        # check that there is no warning on conflictable object
+        # if there is no conflict
+
+        # check that in this case recover button is accessible
+        # check that recovered object is actually being restored
+
+        # make conflict whcih is solvable by user
+        # check that there is a link to conflict
+        # check that there is no recover button
+        pass
```

Comment:
Probably should use self.fail() here so that we don't forget to write this test?